zk_toolbox/crates/zk_inception/src/commands/chain/args/create.rs

```diff
@@ -73,32 +75,48 @@ impl ChainCreateArgs {
             .unwrap_or_else(|| Prompt::new(MSG_CHAIN_NAME_PROMPT).ask());
         chain_name = slugify!(&chain_name, separator = "_");
 
-        let chain_id = self.chain_id.unwrap_or_else(|| {
-            Prompt::new(MSG_CHAIN_ID_PROMPT)
-                .default(&(L2_CHAIN_ID + number_of_chains).to_string())
-                .ask()
+        let chain_id = self
+            .chain_id
+            .map(|v| {
+                (v == 0)
+                    .then(|| L2_CHAIN_ID + number_of_chains)
+                    .unwrap_or(v)
+            })
+            .unwrap_or_else(|| {
+                Prompt::new(MSG_CHAIN_ID_PROMPT)
+                    .default(&(L2_CHAIN_ID + number_of_chains).to_string())
+                    .ask()
+            });
+
+        // NOTE: should check if the option submitted is compatible with the network, but with the
+        // current function signature we can't return an error
```

Comment:
In the case of an interactive execution we disable localhost wallets for external networks, while we can't do the same here because the function is not allowed to fail. Addressing this and the other comments in PR #2579

Reply:
Addressed in #2579